### PR TITLE
Payment Vouchers should be a first class type issue#2527

### PR DESCRIPTION
**Pull request by jscode017**
**What's in this PR?**
    move paymentVoucher to /types
    modify all paymentbroker.PaymentVoucher to types.PaymentVoucher in all files
    create a simple test for PaymentVoucher which encodes then decodes a PaymentVoucher and check 
    if it is equal to the original one

### CODE REVIEW

**Comment by jscode017** [comment]
sorry, I am not so familiar will gofmt.
Sorry again for the inconvenience a caused

**Comment by jscode017** [comment]
I do not know why failed dht_daemon_test.go? Changing payment Vouchers to ./types do not seems to related with dht_daemon?
Can anyone provide some help please.

File: types/payment_voucher_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/address"
+)
+
+func TestEncodeThenDecode(t *testing.T) {
```
Comment (travisperson) [inline types/payment_voucher_test.go]: Can you give this test a name that related it to the payment channel? The scope of this file is shared with the entire `types` package. Something like `TestPaymentVoucherEncodingRoundTrip`.

File: types/payment_voucher_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	rawPaymentVoucher, err := paymentVoucher.Encode()
+	require.NoError(err)
+	DecodedPaymentVoucher, err := DecodeVoucher(rawPaymentVoucher)
+	require.NoError(err)
+
+	assert.Equal((*paymentVoucher).Channel, DecodedPaymentVoucher.Channel)
+	assert.Equal((*paymentVoucher).Payer, DecodedPaymentVoucher.Payer)
+	assert.Equal((*paymentVoucher).Target, DecodedPaymentVoucher.Target)
+	assert.Equal((*paymentVoucher).Amount, DecodedPaymentVoucher.Amount)
+	assert.Equal((*paymentVoucher).ValidAt, DecodedPaymentVoucher.ValidAt)
```
Comment (travisperson) [inline types/payment_voucher_test.go]: Nit: I don't believe the deference here is required, and `D` at the front of `DecodedPaymentVoucher` doesn't need to be capitalized.

Other wise looks good to me

**Comment by jscode017** [comment]
Thanks @travisperson , I change the function name to TestPaymentVoucherEncodingRoundTrip and uncapitalized the D.

**Pull request merged**